win32ss/gdi/eng/gradient.c

```diff
@@ -254,14 +254,12 @@ IntEngGradientFillRect(
   FINITCOL(linefrom, lineto, 0); \
   FINITCOL(linefrom, lineto, 1); \
   FINITCOL(linefrom, lineto, 2); \
-  for(g = sx[linefrom]; g != sx[lineto]; g += gxi) \
+  for(g = sx[linefrom] - 1; g != sx[lineto] + gxi; g += gxi) \
```

Comment:
This '-1' in `sx[linefrom] - 1` should always be `-1` ? Not something like `- gxi` ?